qt/aqt/progress.py

```diff
@@ -47,16 +47,61 @@ def timer(
         If requiresCollection is True, the timer will not fire if the
         collection has been unloaded. Setting it to False will allow the
         timer to fire even when there is no collection, but will still
-        only fire when there is no current progress dialog."""
+        only fire when there is no current progress dialog.
+
+        Issues and alternative
+        ---
+        The created timer will only be destroyed when `parent` is destroyed.
+        This can cause memory leaks, because anything captured by `func` isn't freed either.
+        If there is no QObject that will get destroyed reasonably soon and you have to
+        pass `mw`, you should call `deleteLater()` on the returned QTimer as soon as
+        it's served its purpose, or use `single_shot()`.\n
+        Also note that you may not be able to pass an adequate parent, if you want to
+        make a callback after a widget closes. If you passed that widget, the timer
+        would get destroyed before it could fire.
+        """
 
         if parent is None:
             print_deprecation_warning(
                 "to avoid memory leaks, pass an appropriate parent to progress.timer()"
+                " or use progress.single_shot()"
             )
             parent = self.mw
 
+        qtimer = QTimer(parent)
+        if not repeat:
+            qtimer.setSingleShot(True)
+        qconnect(qtimer.timeout, self._get_handler(func, repeat, requiresCollection))
+        qtimer.start(ms)
+        return qtimer
+
+    def single_shot(
+        self,
+        ms: int,
+        func: Callable[[], None],
+        requires_collection: bool = True,
+    ) -> None:
+        """Create and start a one-of Anki timer. For an alternative and more
+        documentation see `timer()`.
+
+        Issues and alternative
+        ---
+        `single_shot()` cleans itself up, so a passed closure won't leak any memory.
+        However, if `func` references a QObject other than `mw`, which gets deleted before the
+        timer fires, an Exception is raised. To avoid this, either use `timer()` passing
+        that object as the parent, or check in `func` with `sip.isdeleted(object)` if
+        it still exists.\n
```

Comment:
\n? There's another one above too.

Reply:
I wanted a line break in the rendered markdown, but no empty line in the plain text, because that looks disconnected from the heading. But if it causes confusion I better remove it.

Reply:
If one newline makes it clearer, how about two instead :-) Then we can just use standard newlines.